shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/resources/ManagedProcessorStatus.java

```diff
@@ -1,36 +1,24 @@
 package com.redhat.service.smartevents.shard.operator.v2.resources;
 
-import java.util.Objects;
+import java.util.HashSet;
 import java.util.Set;
 
+import com.redhat.service.smartevents.infra.core.models.ManagedResourceStatus;
 import com.redhat.service.smartevents.shard.operator.core.resources.Condition;
+import com.redhat.service.smartevents.shard.operator.core.resources.CustomResourceStatus;
 
-public class ManagedProcessorStatus {
+public class ManagedProcessorStatus extends CustomResourceStatus {
 
-    private Set<Condition> conditions;
-
-    public Set<Condition> getConditions() {
-        return conditions;
-    }
-
-    public void setConditions(Set<Condition> conditions) {
-        this.conditions = conditions;
+    private static Set<Condition> getCreationConditions() {
+        return new HashSet<>();
     }
 
-    @Override
-    public boolean equals(Object o) {
-        if (this == o) {
-            return true;
-        }
-        if (o == null || getClass() != o.getClass()) {
-            return false;
-        }
-        ManagedProcessorStatus that = (ManagedProcessorStatus) o;
-        return Objects.equals(conditions, that.conditions);
+    public ManagedProcessorStatus() {
+        super(getCreationConditions());
     }
 
     @Override
-    public int hashCode() {
-        return Objects.hash(conditions);
+    public ManagedResourceStatus inferManagedResourceStatus() {
```

Comment:
Would it be better to refactor `shard-core` and `shard-v1` to remove this from `shard-v2`?

Reply:
done